### ISSUE TITLE
set doesn't throw error on set when cache is full

### ISSUE THREAD

**Issue by panghalamit** [node-cache/node-cache#146]
```
function throwErrorOnCacheMaxLimit() {
  let nodeCache = new NodeCache({ stdTTL: 20, checkperiod: 5, useClones: false, maxKeys: 1});
  let isSet = nodeCache.set("1", "1");
  try {
    isSet = nodeCache.set("2", "2");
  } catch(err) {
    console.log(err);
  }
  return nodeCache.getStats().keys;
}
```

it returns 2. Shouldn't it return 1 and log the error.

**Comment by erdii** [comment]
`maxKeys` is part of the upcoming 5.x release, thus you can't use it for now :(

**Comment by erdii** [comment]
I have published a 5.0.0-alpha.0 package on npm. You can try it with: `npm install node-cache@alpha`
